client/signup/steps/site-or-domain/index.jsx

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import StepWrapper from 'signup/step-wrapper';
+import SignupActions from 'lib/signup/actions';
+import Card from 'components/card';
+// TODO: `design-type-with-store`, `design-type`, and this component could be refactored to reduce redundancy
+import BlogImage from 'signup/steps/design-type-with-store/blog-image';
+import PageImage from 'signup/steps/design-type-with-store/page-image';
+
+export default class SiteOrDomain extends Component {
+	getChoices() {
+		return [
+			{
+				type: 'page',
+				label: 'Start a new site',
+				image: <BlogImage />
+			},
+			{
+				type: 'domain',
+				label: 'Just buy a domain',
+				image: <PageImage />
+			},
+		];
+	}
+
+	renderChoices() {
+		return (
+			<div className="site-or-domain__choices">
+				{ this.getChoices().map( ( choice ) => (
+					<Card className="site-or-domain__choice" key={ choice.type }>
+						<a href="#" onClick={ ( event ) => this.handleClickChoice( event, choice.type ) }>
+							{ choice.image }
+							<h2 className="site-or-domain__label">{ choice.label }</h2>
+						</a>
+					</Card>
+				) ) }
+			</div>
+		);
+	}
+
+	handleClickChoice( event, designType ) {
+		event.preventDefault();
+
+		const { stepName, goToStep, goToNextStep } = this.props;
+
+		SignupActions.submitSignupStep( { stepName }, [], { designType } );
+
+		if ( designType === 'domain' ) {
+			// we can skip the next two steps in the `domain-first` flow if the
+			// user is only purchasing a domain
+			SignupActions.submitSignupStep( { stepName: 'themes', wasSkipped: true }, [], { theme: 'pub/twentysixteen' } );
```

Comment:
@drewblaisdell Are we going to update `theme` name in this place? I though about that after looking at another task :)